--- main/codec_builtin.c.orig	2022-02-10 15:18:11.739335845 -0300
+++ main/codec_builtin.c	2022-02-10 15:19:57.718923445 -0300
@@ -146,7 +146,7 @@
 	.type = AST_MEDIA_TYPE_AUDIO,
 	.sample_rate = 8000,
 	.minimum_ms = 10,
-	.maximum_ms = 150,
+	.maximum_ms = 120,
 	.default_ms = 20,
 	.minimum_bytes = 80,
 	.samples_count = ulaw_samples,
@@ -160,7 +160,7 @@
 	.type = AST_MEDIA_TYPE_AUDIO,
 	.sample_rate = 8000,
 	.minimum_ms = 10,
-	.maximum_ms = 150,
+	.maximum_ms = 120,
 	.default_ms = 20,
 	.minimum_bytes = 80,
 	.samples_count = ulaw_samples,
@@ -419,7 +419,7 @@
 	.type = AST_MEDIA_TYPE_AUDIO,
 	.sample_rate = 8000,
 	.minimum_ms = 10,
-	.maximum_ms = 230,
+	.maximum_ms = 220,
 	.default_ms = 20,
 	.minimum_bytes = 10,
 	.samples_count = g729_samples,
